### PR TITLE
split nw4r/snd_dvdSoundArchive, eggAudioArcPlayerMgr

### PR DESCRIPTION
**Pull request by terorie**

### CODE REVIEW

File: source/egg/audio/eggAudioArcPlayerMgr.cpp [hunk 1]
```diff
@@ -0,0 +1,867 @@
...
+  mtctr r12;
+  bctrl;
+  mr r0, r3;
+lbl_80210734:
+  mr r3, r0;
+  lwz r0, 0x14(r1);
+  mtlr r0;
+  addi r1, r1, 0x10;
+  blr;
+  stwu r1, -0x20(r1);
```
Comment (terorie) [inline source/egg/audio/eggAudioArcPlayerMgr.cpp:221]: is this a stack frame / new function?

File: source/nw4r/snd/snd_dvdSoundArchive.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  // PAL: 0x80091010..0x80091090
+  ~DvdSoundArchive();
+
+  // PAL: 0x80091090..0x800911d0
+  bool Open(const char* path);
+  // PAL: 0x800911d0..0x80091210
+  void Close();
+
+  // PAL: 0x80091380..0x80091390
+  int detail_GetRequiredStreamBufferSize__Q34nw4r3snd15DvdSoundArchiveFv();
```
Comment (terorie) [inline source/nw4r/snd/snd_dvdSoundArchive.hpp:23]: typo

File: source/egg/audio/eggAudioArcPlayerMgr.cpp (hunk 1 above)
Comment (riidefi) [reply source/egg/audio/eggAudioArcPlayerMgr.cpp:221]: Yep. Line 180 for example as well. There are many functions in that one assembly blob. IDA/Ghidra can detect those boundaries.

Comment (terorie) [reply source/egg/audio/eggAudioArcPlayerMgr.cpp:239]: Should be fixed now. Will try to demangle next

File: source/egg/audio/eggAudioArcPlayerMgr.cpp [hunk 3]
```diff
@@ -0,0 +1,961 @@
...
+extern UNKNOWN_FUNCTION(close__Q23EGG7CntFileFv);
+// PAL: 0x802150a0
+extern UNKNOWN_FUNCTION(readData__Q23EGG7CntFileFPvll);
+}
+
+// Symbol: __ct__Q23EGG9ArcPlayerFPQ34nw4r3snd9SoundHeap
+// PAL: 0x80210590..0x80210624
+MARK_BINARY_BLOB(__ct__Q23EGG9ArcPlayerFPQ34nw4r3snd9SoundHeap, 0x80210590,
+                 0x80210624);
+asm ArcPlayer::ArcPlayer(nw4r::snd::SoundHeap* heap) {
```
Comment (terorie) [inline source/egg/audio/eggAudioArcPlayerMgr.cpp:83]: compiler ICE

**Pull request closed without merging**